sklearn/preprocessing/_encoders.py

```diff
@@ -363,6 +486,201 @@ def _compute_drop_idx(self):
                              zip(self.drop, self.categories_)],
                             dtype=np.object)
 
+    def _infrequent_enabled(self):
+        """Infrequent category is enabled."""
+        if self.max_categories is not None and self.max_categories > 1:
+            return True
+        if (isinstance(self.min_frequency, numbers.Integral)
+                and self.min_frequency > 1):
+            return True
+        if (isinstance(self.min_frequency, numbers.Real)
+                and 0.0 < self.min_frequency < 1.0):
+            return True
+        return False
+
+    def _identify_infrequent(self, category_count, n_samples, col_idx):
+        """Compute the infrequent indicies based on max_categories and
+        min_frequency.
+
+        Parameters
+        ----------
+        category_count : ndarray of shape (n_cardinality,)
+            Category counts.
+
+        n_samples : int
+            Number of samples.
+
+        col_idx : int
+            Index of the current category. Only used for the error message.
+
+        Returns
+        -------
+        output : ndarray of shape (n_infrequent_categories,) or None
+            If there are infrequent categories, indicies of infrequent
+            categories. Otherwise None.
+        """
+        # categories with no count are infrequent
+        infrequent_mask = category_count == 0
```

Comment:
Do we need this? this is going to be redundant with the ``min_frequency`` computation, right? can't we just do ``=`` there? ``self.min_frequency=1`` by default which is equivalent to this if it's integer, and if it's float it's required to be > 0

Reply:
Agreed